modules/xfem/include/userobjects/GeometricCutUserObject.h

```diff
@@ -25,6 +25,23 @@ struct CutEdge
   unsigned int host_side_id;
 };
 
+/**
+ * To allow the use of std::set<CutEdge>, the operator of < has to be overloaded for CutEdge.
+ */
+inline bool
+operator<(const CutEdge & lhs, const CutEdge & rhs)
+{
+  unsigned int l1 = lhs.id1;
+  unsigned int l2 = lhs.id2;
+  if (l1 > l2)
+    std::swap(l1, l2);
+  unsigned int r1 = rhs.id1;
+  unsigned int r2 = rhs.id2;
+  if (r1 > r2)
+    std::swap(r1, r2);
+  return l1 < r1 || l2 < r2;
```

Comment:
I hope I understood your comment correctly.  With the following changes, now (1,3)<(2,2) and (1,10)<(2,3).

I changed from: 
  return l1 < r1 || l2 < r2;

to:
  if (l1 != r1)
    return l1 < r1;
  else
    return l2 < r2;

Reply:
> if (l1 != r1)
return l1 < r1;
else
return l2 < r2;

Yes, that makes sense, except you need to get rid of all the swap logic. You only need the comparison logic.

Reply:
@permcody The swap logic is to make an edge with two nodes unique in comparison.  For example, (1,3) and (3,1) are geometrically the same edge.  We probably don't want the overloaded operator to give (1,3)<(2,4) true and (3,1)<(2,4) false. I used the swap logic to avoid these.  Does this make sense?

Reply:
That makes sense but it terms of storing items in a map I'm not sure it really matters.

I'm more concerned about confusing the map. If during the comparison for inserting an item you tell the map to go into the left tree for one case and into the right tree for another sometime you can insert items in the wrong place making the lookup not work like you might expect. That might not happen here, I'd have to sit down and really work through a few cases, but I'm concerned you might fall into one of those cases. If you think it's correct, then by all means, leave it.

Reply:
What you wrote in the email makes sense to me.  I have removed the swap logic from this operator, and added it to my code where all CutEdges are stored.  This will make this operator more universal (not potentially confusing the map as you described in the email).  And it meets my needs too (avoid storing (1,3) and (3,1) for the same edge). Below is the updated code. Thanks!

inline bool
operator<(const CutEdge & lhs, const CutEdge & rhs)
{
  if (lhs.id1 != rhs.id1)
    return lhs.id1 < rhs.id1;
  else
    return lhs.id2 < rhs.id2;
}